### PR TITLE
[BUILD] Revise Kyuubi Spark engine shaded

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

This PR revises the shaded rule of spark engine module, especially to make sure that netty native libs is shaded properly.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

```
(kyuubi) ➜  apache-kyuubi git:(spark-engine-shade) tree externals/kyuubi-spark-sql-engine/target/unpacked -L 5 | grep -v class
externals/kyuubi-spark-sql-engine/target/unpacked
├── META-INF
│   ├── LICENSE
│   ├── MANIFEST.MF
│   ├── NOTICE
│   ├── io.netty.versions.properties
│   ├── native
│   │   ├── liborg_apache_kyuubi_shade_netty_transport_native_epoll_aarch_64.so
│   │   └── liborg_apache_kyuubi_shade_netty_transport_native_epoll_x86_64.so
│   ├── services
│   │   ├── org.apache.kyuubi.shade.io.grpc.LoadBalancerProvider
│   │   ├── org.apache.kyuubi.shade.io.grpc.ManagedChannelProvider
│   │   ├── org.apache.kyuubi.shade.io.grpc.NameResolverProvider
│   │   ├── org.apache.kyuubi.shade.io.grpc.ServerProvider
│   │   ├── org.apache.kyuubi.shade.io.vertx.core.spi.launcher.CommandFactory
│   │   ├── org.apache.spark.status.AppHistoryServerPlugin
│   │   └── reactor.blockhound.integration.BlockHoundIntegration
│   ├── versions
│   │   └── 11
│   │       └── io
│   │           └── vertx
│   └── vertx
│       └── vertx-version.txt
├── kyuubi-version-info.properties
├── log4j2-defaults.xml
├── org
│   └── apache
│       ├── kyuubi
│       │   ├── cli
│       │   ├── config
│       │   │   └── internal
│       │   ├── engine
│       │   │   └── spark
│       │   ├── events
│       │   │   └── handler
│       │   ├── ha
│       │   │   └── client
│       │   ├── operation
│       │   │   ├── log
│       │   │   └── meta
│       │   ├── reflection
│       │   ├── service
│       │   │   └── authentication
│       │   ├── session
│       │   ├── shade
│       │   │   ├── android
│       │   │   ├── com
│       │   │   ├── io
│       │   │   ├── net
│       │   │   └── org
│       │   └── util
│       └── spark
│           ├── api
│           │   └── python
│           ├── kyuubi
│           └── ui
└── python
    ├── execute_python.py
    └── kyuubi_util.py

40 directories, 211 files
```

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
There are incompatible changes in Netty for early Spark release, then the shade of Netty is required.
```
22/11/07 14:01:10 ERROR ManagedChannelImpl: [Channel<1>: (ip:///localhost:49165)] Uncaught exception in the SynchronizationContext. Panic!
java.lang.NoSuchMethodError: io.netty.buffer.PooledByteBufAllocator.<init>(ZIIIIIIZ)V
	at org.apache.kyuubi.shade.io.grpc.netty.Utils.createByteBufAllocator(Utils.java:180)
	at org.apache.kyuubi.shade.io.grpc.netty.Utils.access$000(Utils.java:75)
	at org.apache.kyuubi.shade.io.grpc.netty.Utils$ByteBufAllocatorPreferDirectHolder.<clinit>(Utils.java:98)
	at org.apache.kyuubi.shade.io.grpc.netty.Utils.getByteBufAllocator(Utils.java:148)
	at org.apache.kyuubi.shade.io.grpc.netty.NettyClientTransport.start(NettyClientTransport.java:233)
```

**Comment by pan3793** [comment]
ping @hddong, it's ready for reviewing and testing

File: externals/kyuubi-spark-sql-engine/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -23,269 +23,5 @@ Copyright 2019 The Apache Software Foundation
 Apache Thrift
 Copyright 2006-2010 The Apache Software Foundation.
 
-Google Guava Version 18.0
-* Copyright (C) 2009 The Guava Authors
```
Comment (hddong) [inline externals/kyuubi-spark-sql-engine/src/main/resources/META-INF/NOTICE:27]: Accidentally deleted? Seems Guava still be include.

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/resources/META-INF/NOTICE:27]: The recent version Guava does not ship NOTICE in the jar

Comment (hddong) [reply externals/kyuubi-spark-sql-engine/src/main/resources/META-INF/NOTICE:27]: Get, LGTM, I will have a test locally.

**Review by hddong — APPROVED** [review]

**Comment by hddong** [comment]
Checked locally, it works as expected.

**Comment by pan3793** [comment]
Thanks, merging to master

**Comment by pan3793** [comment]
Cherry picked to branch-1.6 as user reported that jetcd class conflict w/ TiSpark

**Pull request closed without merging**